src/BPDual.jl

```diff
@@ -100,6 +100,13 @@ function bpdual(
     # ------------------------------------------------------------------
     m, n = size(A)
 
+
+    work = rand(n)
+    work2 = rand(n)
+    work3 = rand(n)
+    work4 = rand(n)
+    work5 = rand(m)
```

Comment:
Presumably `rand` is being used to initialize these work arrays, and random numbers aren't really required? If so, replace with undefined allocations of the correct type.